docs/migration.md

```diff
@@ -22,6 +22,133 @@ We will aim to provide:
 
 ## Migration Log
 
+### Auth overhaul - April 20, 2024
+
+This release overhauls and simplifies our authentication and authorization systems.
+If you are you using Chroma's built-in auth system, you will need to update your configuration and
+any code you wrote to implement your own authentication or authorization providers.
+This change is mostly to pay down some of Chroma's technical debt and make future changes easier,
+but it also changes and simplifies user configuration.
+If you are not using Chroma's built-in auth system, you do not need to take any action.
+
+Previously, Chroma's authentication and authorization relied on many objects with many configuration options, including:
+
+- `chroma_server_auth_provider`
+- `chroma_server_auth_configuration_provider`
+- `chroma_server_auth_credentials_provider`
+- `chroma_client_auth_credentials_provider`
+- `chroma_client_auth_protocol_adapter`
+
+and others.
+
+We have consolidated all of this complexity into three classes:
+
+- `ClientAuthProvider`
+- `ServerAuthenticationProvider`
+- `ServerAuthorizationProvider`
+
+`ClientAuthProvider`s are now responsible for their own configuration and credential management. Credentials can be given to them with the `chroma_client_auth_credentials` setting.
+`ServerAuthenticationProvider`s are responsible for turning a request's authorization information into a `UserIdentity` containing any information necessary to make an authorization decision. They are now responsible for their own configuration and credential management. Configured via the `chroma_server_authn_credentials` and `chroma_server_authn_credentials_file` settings.
```

Comment:
dumb_reader question what is credentials vs credentials_file?

Reply:
Great question! I realized this should actually go in the Usage Guide which I missed before. Updated the usage guide and pointed to it from here.